src/PrestaShopBundle/Form/Admin/Improve/International/Translations/TranslationsSettingsFormHandler.php

```diff
@@ -75,18 +75,16 @@ public function __construct(
      */
     public function getForm()
     {
-        foreach ($this->formTypes as $formName => $formType) {
-            $this->formBuilder->add($formName, $formType);
-        }
+        $form = $this->formFactory->createNamed('form', $this->form);
 
         $this->hookDispatcher->dispatchWithParameters(
             "action{$this->hookName}Form",
             [
-                'form_builder' => $this->formBuilder,
+                'form_builder' => $form,
```

Comment:
The form builder is also used in hooks in other classes in the core `FormHandler` `(Indentifiable)FormBuilder` So it would not be consistent (and not handy) if translation handler is the only to use the form directly

Reply:
The problem is, we don't have any form builder yet and we should never have used the `form_builder` here.  Ping @eternoendless

Reply:
Maybe you can do something similar to the form builder for identifiable builder: https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Form/IdentifiableObject/Builder/FormBuilder.php#L120

```php
$formBuilder = $this->formFactory->createBuilder($formType, $data, $options);

        $this->hookDispatcher->dispatchWithParameters('action' . Container::camelize($formBuilder->getName()) . 'FormBuilderModifier', [
            'form_builder' => $formBuilder,
            'data' => &$data,
            'id' => $id,
        ]);

        return $formBuilder->getForm();
```

Reply:
@PierreRambaud We can duplicate Symfony `createNamedBuilder()` function and use it instead of `CreateNamed()` which allows us to call `createBuilder()` ourselves. This allows us to fix the issue with token AND keep sending the `form_builder` in the hook. Wdyt ?

Reply:
Thanks @matks it's done :)